libdnf/base/log_event.cpp

```diff
@@ -121,7 +121,12 @@ std::string LogEvent::to_string(
             if (additional_data->size() != 1) {
                 throw std::invalid_argument("Incorrect number of elements for ALREADY_INSTALLED");
             }
-            return ret.append(utils::sformat(_("Package \"{}\" is already installed."), *additional_data->begin()));
+            if (action == GoalAction::REASON_CHANGE) {
+                return ret.append(utils::sformat(
+                    _("Package \"{}\" is already installed with correct reason."), *additional_data->begin()));
```

Comment:
That's nice. I suggest to improve the report with information about `reason`. A minor modification needs to be made to "goal.cpp", which I also suggested.

`_("Package \"{}\" is already installed with correct reason."), *additional_data->begin()));` -> `_("Package \"{}\" is already installed with reason \"{}\"."), *spec, *additional_data->begin()));`